src/ripple/basics/spinlock.h

```diff
@@ -75,8 +117,16 @@ class packed_spinlock
     packed_spinlock&
     operator=(packed_spinlock const&) = delete;
 
+    /** A single spinlock packed inside the specified atomic
+
+        @param lock The atomic integer inside which the spinlock is packed.
+        @param index The index of the spinlock this object acquires.
+
+        @note For performance reasons, you should strive to have `lock` be
+              on a cacheline by itself.
+     */
     packed_spinlock(std::atomic<T>& lock, int index)
-        : bits_(lock), mask_(1 << index)
+        : bits_(lock), mask_(detail::masks<T>[index])
```

Comment:
I'm skeptical this makes a difference. I don't object, but I prefer the bitshift (unless you have a benchmark that shows this matters).

Reply:
It doesn't _really_ matter, but it is cool, isn't it? :)